### PR TITLE
[stable/airflow] update condition

### PR DESCRIPTION
**Pull request by MahaGamal**
#### What this PR does / why we need it:
Revert this PR (#15667).  and fix condition

### CODE REVIEW

File: stable/airflow/templates/_helpers.tpl [hunk 1]
```diff
@@ -97,7 +97,7 @@ See script/entrypoint.sh in that repo for more info.
 The key names for postgres and redis are fixed, which is consistent with the subcharts.
 */}}
 {{- define "airflow.mapenvsecrets" }}
-  {{- if .Values.postgresql.enabled }}
+  {{- if .Values.postgresql.existingSecret }}
```
Comment (ms32035) [inline stable/airflow/templates/_helpers.tpl]: This should be 2 lines down I believe

Comment (MahaGamal) [reply stable/airflow/templates/_helpers.tpl]: the file structure don't make 2 line down after define

Comment (ms32035) [reply stable/airflow/templates/_helpers.tpl]: I meant it should only affect POSTGRES_PASSWORD not POSTGRES_USER

Comment (gsemet) [reply stable/airflow/templates/_helpers.tpl]: I agree it should be only for the password (example [here](https://github.com/helm/charts/blob/a63082fe8cbf6bac2add8151c0775c0e16ce3244/stable/sentry/templates/hooks/user-create.job.yaml#L47):
```yaml
{{- define "airflow.mapenvsecrets" }}
  - name: POSTGRES_USER
    value: {{ default "postgres" .Values.postgresql.postgresUser | quote }}
{{- if .Values.postgresql.existingSecret }}
- name: POSTGRES_PASSWORD
      name: {{ default (include "airflow.postgresql.fullname" .)
      key: postgres-password
{- end }}

Comment (MahaGamal) [reply stable/airflow/templates/_helpers.tpl]: I agree, but if i use external Postgres must be using `.Values.postgresql.postgresUser` user

Comment (gsemet) [reply stable/airflow/templates/_helpers.tpl]: yes you can define your user for your external database in `.Values.postgresql.postgresUser`.

Comment (MahaGamal) [reply stable/airflow/templates/_helpers.tpl]: I think this is not true when set `.Values.postgresql.enable` false, I assume that i don't use any variable under postgres.

Comment (ms32035) [reply stable/airflow/templates/_helpers.tpl]: dc6a359 seems ok. That should also work in a case when you have an external Postres with a user but no password (AWS IAM authentication for example)

Comment (MahaGamal) [reply stable/airflow/templates/_helpers.tpl]: @gsemet approve please

**Review by gsemet — APPROVED** [review]
/lgtm

Thank !

**Pull request merged**